sklearn/utils/metaestimators.py

```diff
@@ -19,24 +19,35 @@ class _IffHasAttrDescriptor(object):
     of the base object or the first found delegate does not have an attribute
     ``attribute_name``.
 
+    If ``disable_on_multimetric`` is set to ``True``
+
     This allows ducktyping of the decorated method based on
     ``delegate.attribute_name``. Here ``delegate`` is the first item in
     ``delegate_names`` for which ``hasattr(object, delegate) is True``.
 
     See https://docs.python.org/3/howto/descriptor.html for an explanation of
     descriptors.
     """
-    def __init__(self, fn, delegate_names, attribute_name):
+    def __init__(self, fn, delegate_names, attribute_name,
+                 disable_on_multimetric):
```

Comment:
I don't think we should be adding something so specialised. But I'm not sure we need to add anything at all. Surely wherever `best_estimator_` has a value, we should be delegating.